src/listoper.c

```diff
@@ -1006,42 +1006,44 @@ static Obj OneMatrix(Obj mat, UInt mut)
       one = ONE_SAMEMUT(zero);
       CheckedMakeImmutable(zero);
       CheckedMakeImmutable(one);
-      ctype = rtype = T_PLIST+IMMUTABLE;
+      cmut = rmut = FALSE;
       break;
 
     case 1:
       zero = ZERO_MUT( ELM_LIST( ELM_LIST( mat, 1 ), 1 ) );
       one = ONE_SAMEMUT(zero);
       if (IS_MUTABLE_OBJ(mat))
         {
-          ctype = T_PLIST;
-          rtype = IS_MUTABLE_OBJ(ELM_LIST(mat, 1)) ? T_PLIST : T_PLIST + IMMUTABLE;
+          cmut = TRUE;
+          rmut = IS_MUTABLE_OBJ(ELM_LIST(mat, 1));
         }
       else
-        ctype = rtype = T_PLIST + IMMUTABLE;
+          cmut = rmut = FALSE;
       break;
 
     case 2:
       zero = ZERO_SAMEMUT( ELM_LIST( ELM_LIST( mat, 1 ), 1 ) );
       one  = ONE( zero );
-      ctype = rtype = T_PLIST;
+      cmut = rmut = TRUE;
       break;
```

Comment:
can this switch gain a default:, or the variables gain default values, just because now falling through this switch if mut isn't 0,1 or 2 is now undefined behaviour. Or, GAP_ASSERT that fact above the switch?:

Reply:
I'm happy now, but you did make a change -- you removed the variables previously being assigned 0 by default, to having no value, which would change behaviour on fall-through!

Reply:
But then again there are just three callers to this static function... and even with the default value, a fall through would still lead to nonsense. And not to forget the other argument which could receive a nonsense argument...